daemon/sdnotify.go

```diff
@@ -2,30 +2,33 @@
 package daemon
 
 import (
-	"errors"
 	"net"
 	"os"
 )
 
-var SdNotifyNoSocket = errors.New("No socket")
-
 // SdNotify sends a message to the init daemon. It is common to ignore the error.
```

Comment:
I suggest the following (yet simpler), leave the signature as-is:

`func SdNotify(state string) error`

and return:

return `nil`: notification supported, data has been sent
return `ErrNotSupported`: notification not supported, where `var ErrNotSupported = errors.New("unsupported")`
return `err`: notification supported, but failure happened (`err != ErrNotSupported`)